### ISSUE TITLE
Problem encoding message when using encodeMessageByTopicName().

### ISSUE THREAD

**Issue by hanjiwoo** [bencebalogh/avro-schema-registry#39]
Hi,

I'm trying to encode message with avro schema using encodeMessageByTopicName() of this library. 
But, when I produce a message to same topic from the second time, it gives me a following error.
(only if trying it only one time or this trying is the first time of several attempts, it doesn't give me a error.)
```
TypeError: getSchemaByTopicName(...)(...).then is not a function
message:"getSchemaByTopicName(...)(...).then is not a function"
stack: "TypeError: getSchemaByTopicName(...)(...).then is not a function at Object.encodeMessageByTopicName
```

So, I tried to debug your code and found a code that look like a little bit wrong.

In line 60 of `encode-function.js`, return just a returned value of cache.getById(), that is  parsedSchema, not a Promise in spite of received 'then' in line 70.

```
 59  if (schemaId) {
 60    return registry.cache.getById(schemaId);
 61  }
 ...
 70  const byTopicName = (registry) => (topic, msg, parseOptions = null) => getSchemaByTopicName(registry)(topic, parseOptions).then(({id, parsedSchema}) => encodeMessage(msg, id)(parsedSchema))
```

So, there is a way to solve this problem?

> when I fixed a code like this, my code've run normally.
> ```
> return Promise.resolve({ id: schemaId, parsedSchema: registry.cache.getById(schemaId) })
> ```

**Comment by bencebalogh** [comment]
Hi
I think you're right. I started fixing this, but since the original inception of this lib a few unexpected requests came so I'll do a major refactor to make fixing this and #36 easier and also adding the features in #28

**Comment by hanjiwoo** [comment]
Ok, thanks bence 😃 

By the way, I saw pull request #31 for issue #28 and it already fix this issue via commit [6f7e5aa](https://github.com/bencebalogh/avro-schema-registry/pull/31/commits/6f7e5aabd36fee611e6d2df7279d2955c485dc25)!!
So if the pull request is merged, I think this issue would have been solved too.

**Comment by dlrudwns135** [comment]
i have same error

**Comment by exitsmoker** [comment]
I look forward to merge this without sleeping :)

**Comment by bencebalogh** [comment]
just for visibility: I've opened #42 , I am checking at the moment to make sure nothing it works as expected. It'll fix this issue + make future changes more simple

**Comment by bencebalogh** [comment]
should be fixed in 2.1.0
